### ISSUE TITLE
Local git repo

### ISSUE THREAD

**Issue by geek1011** [abiosoft/caddy-git#85]
caddy-git does not work properly with a path to a bare git repo as the repo url. This is useful for using caddy to serve git repos on the same server.

**Comment by abiosoft** [comment]
Can you give examples ?

**Comment by geek1011** [comment]
I use Gitea on the same server as caddy, and I want to serve a private repo from the bare repo in the Gitea data folder.

**Comment by abiosoft** [comment]
Yeah, that makes sense. I'm actually referring to example repo URL that is failing.

**Comment by geek1011** [comment]
/srv/gitea/repositories/geek1011/whatever.git

**Comment by abiosoft** [comment]
why not `http://127.0.0.1/whatever.git` ?

**Comment by geek1011** [comment]
Because the repo is private, I don't want to put my ssh key on the server, and gitea is proxied through caddy (so it would deadlock).

**Comment by abiosoft** [comment]
Now I understand better. But that will be limited to long polling, webhooks can't work.

**Comment by geek1011** [comment]
Yes, I am aware of that.

**Comment by lesion** [comment]
same issue here.
I'm using [gogs](https://gogs.io/) to serve git repos proxied by caddy itself.
With this configuration I cannot use this `git` directive to pull from a git repo served by caddy itself :) and there's no `file://` support neither so I cannot find a workaround.

**Comment by csos95** [comment]
I'm having a similar issue.  
I use caddy to proxy my gitlab installation and I cannot use this plugin because it tries to pull the repo before gitlab is available through caddy.  
Would it be possible to add an option to not pull the repos at launch?

**Comment by abiosoft** [comment]
If this would resolve the issue then yeah, it can be added.
> Would it be possible to add an option to not pull the repos at launch?

**Comment by csos95** [comment]
I'm not sure if it will resolve the initial issue of wanting to pull a local repo from the filesystem, but it would solve the issue @lesion and I have of wanting to pull from a gogs/gitlab install on the same server.
